sklearn/model_selection/tests/test_split.py

```diff
@@ -891,27 +891,37 @@ def test_leave_group_out_changing_groups():
 
 def test_leave_one_p_group_out_error_on_fewer_number_of_groups():
     X = y = groups = np.ones(0)
-    assert_raise_message(ValueError, "Found array with 0 sample(s)", next,
-                         LeaveOneGroupOut().split(X, y, groups))
+    msg = re.escape("Found array with 0 sample(s)")
+    with pytest.raises(ValueError, match=msg):
+        next(LeaveOneGroupOut().split(X, y, groups))
+
     X = y = groups = np.ones(1)
-    msg = ("The groups parameter contains fewer than 2 unique groups ({}). "
-           "LeaveOneGroupOut expects at least 2.").format(groups)
-    assert_raise_message(ValueError, msg, next,
-                         LeaveOneGroupOut().split(X, y, groups))
+    msg = re.escape(
+        "The groups parameter contains fewer than 2 unique groups ({}). "
+        "LeaveOneGroupOut expects at least 2.".format(groups)
+    )
```

Comment:
Since we support only Python 3.6+ we can use an f-string instead of the `format` API:

```suggestion
    msg = re.escape(
        f"The groups parameter contains fewer than 2 unique groups"
        f"({groups}). LeaveOneGroupOut expects at least 2."
    )
```